pokete_classes/providers.py

```diff
@@ -27,12 +44,28 @@ def get_attack(self, fightmap, enem):
             anem: The enemy Provider"""
         raise NotImplementedError
 
+    def greet(self, fightmap):
+        """Outputs a greeting text at the fights start:
+        ARGS:
+            fightmap: fightmap object"""
+        raise NotImplementedError
```

Comment:
You're raising a `NotImplementedError` here, which makes me think this is an abstract method that needs to be overloaded by the child classes. This is a nice use case for the built-in `abc` module. Consider:

```py
from abc import ABC

class Provider(ABC):
    ...
    @abstractmethod
     def greet(self, fightmap):
         pass
```
If you define a child class that does *not* define the abstract method(s), then you get an error when trying to instantiate it, which is nice.

Reply:
That's what I was trying to achieve.